internal/characters/candace/burst.go

```diff
@@ -0,0 +1,135 @@
+package candace
+
+import (
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/keys"
+	"github.com/genshinsim/gcsim/pkg/core/player/character"
+	"github.com/genshinsim/gcsim/pkg/core/player/weapon"
+	"github.com/genshinsim/gcsim/pkg/modifier"
+)
+
+var burstFrames []int
+
+const (
+	burstHitmark = 33
+	burstKey     = "candace-burst"
+	waveHitmark  = 1
+)
+
+func init() {
+	burstFrames = frames.InitAbilSlice(51)
+	burstFrames[action.ActionAttack] = 50
+	burstFrames[action.ActionDash] = 50
+	burstFrames[action.ActionJump] = 50
+	burstFrames[action.ActionSwap] = 49
+}
+
+func (c *char) Burst(p map[string]int) action.ActionInfo {
+	c.waveCount = 0
+	ai := combat.AttackInfo{
+		ActorIndex:         c.Index,
+		Abil:               "Sacred Rite: Wagtail's Tide (Q)",
+		AttackTag:          combat.AttackTagElementalBurst,
+		ICDTag:             combat.ICDTagNone,
+		ICDGroup:           combat.ICDGroupDefault,
+		StrikeType:         combat.StrikeTypeBlunt,
+		Element:            attributes.Hydro,
+		Durability:         25,
+		FlatDmg:            burstDmg[c.TalentLvlBurst()] * c.MaxHP(),
+		HitlagFactor:       0.01,
+		CanBeDefenseHalted: true,
+	}
+	c.Core.QueueAttack(
+		ai,
+		combat.NewCircleHit(c.Core.Combat.Player(), 1, false, combat.TargettableEnemy),
+		burstHitmark,
+		burstHitmark,
+	)
+
+	duration := 540
+	if c.Base.Cons >= 1 {
+		duration = 720
+	}
+
+	// timer starts at hitmark
+	c.QueueCharTask(func() {
+		for _, char := range c.Core.Player.Chars() {
+			char.AddAttackMod(character.AttackMod{
+				Base: modifier.NewBaseWithHitlag(burstKey, duration),
+				Amount: func(atk *combat.AttackEvent, _ combat.Target) ([]float64, bool) {
+					if atk.Info.AttackTag != combat.AttackTagNormal {
+						return nil, false
+					}
+					if atk.Info.Element == attributes.Physical || atk.Info.Element == attributes.NoElement {
+						return nil, false
+					}
+					m := make([]float64, attributes.EndStatType)
+					m[attributes.DmgP] = 0.2
+					return m, true
+				},
+			})
+			switch char.Weapon.Class {
+			case weapon.WeaponClassClaymore,
+				weapon.WeaponClassSpear,
+				weapon.WeaponClassSword:
+				c.Core.Player.AddWeaponInfuse(
```

Comment:
I suspect this is not how the burst buff/infusion is supposed to be applied. Do we know how often this buff refresh, if at all?

Reply:
See https://github.com/genshinsim/gcsim/pull/959#discussion_r996185991.